src/client/components/pinboard-tile/pinboard-tile.tsx

```diff
@@ -0,0 +1,255 @@
+import { DragManager } from "../../utils/drag-manager/drag-manager";
+import { reportError } from "../../utils/error-reporter/error-reporter";
+import { Loader } from "../loader/loader";
+import { QueryError } from "../query-error/query-error";
+import { SearchableTile } from "../searchable-tile/searchable-tile";
+import { PinboardDataset } from "./pinboard-dataset";
+import { pinboardIcons } from "./pinboard-icons";
+import "./pinboard-tile.scss";
+import { EditMode, RowsMode } from "./utils/edit-mode";
+import { isClauseEditable } from "./utils/is-clause-editable";
+import { isDimensionPinnable } from "./utils/is-dimension-pinnable";
+import { makeQuery } from "./utils/make-query";
+import { isPinnableClause, PinnableClause } from "./utils/pinnable-clause";
+import { equalParams, QueryParams } from "./utils/query-params";
+import { shouldFetchData } from "./utils/should-fetch";
+import { tileStyles } from "./utils/tile-styles";
+
+export class PinboardTileProps {
+  essence: Essence;
+  clicker: Clicker;
+  dimension: Dimension;
+  timekeeper: Timekeeper;
+  sortOn: SortOn;
+  onClose: Fn;
+}
+
+export interface PinboardTileState {
+  searchText: string;
+  showSearch: boolean;
+  datasetLoad: DatasetLoad;
+}
+
+const noMeasureError = new Error("No measure selected");
+
+export class PinboardTile extends React.Component<PinboardTileProps, PinboardTileState> {
+
+  state: PinboardTileState = {
+    searchText: "",
+    showSearch: false,
+    datasetLoad: this.props.sortOn ? loading : error(noMeasureError)
+  };
+
+  private loadData(params: QueryParams) {
+    if (!params.sortOn) {
+      this.setState({ datasetLoad: error(noMeasureError) });
+      return;
+    }
+    this.setState({ datasetLoad: loading });
+    this.fetchData(params)
+      .then(loadedDataset => {
+        // TODO: encode it better
+        // null is here when we get out of order request, so we just ignore it
+        if (!loadedDataset) return;
+        this.setState({ datasetLoad: loadedDataset });
+      });
+  }
+
+  private fetchData(params: QueryParams): Promise<DatasetLoad | null> {
+    this.lastQueryParams = params;
+    return this.debouncedCallExecutor(params);
+  }
+
+  private lastQueryParams: Partial<QueryParams> = {};
+
+  private callExecutor = (params: QueryParams): Promise<DatasetLoad | null> => {
+    const { essence: { timezone, dataCube } } = params;
+    return dataCube.executor(makeQuery(params), { timezone })
+      .then((dataset: Dataset) => {
+          // signal out of order requests with null
+          if (!equalParams(params, this.lastQueryParams)) return null;
+          return loaded(dataset);
+        },
+        err => {
+          // signal out of order requests with null
+          if (!equalParams(params, this.lastQueryParams)) return null;
+          reportError(err);
+          return error(err);
+        });
+  };
+
+  private debouncedCallExecutor = debounceWithPromise(this.callExecutor, 500);
```

Comment:
Should we handle debouncing here? Or it should be done by datacube executor?

Reply:
I think that would be too global. Opening filter menu should invoke call right away, for example.